### PR TITLE
Adding Spin Trigger MQTT documentation

### PR DESCRIPTION
**Pull request by tpmccallum**

### CODE REVIEW

File: content/en/docs/spin-trigger-mqtt/installation/_index.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+## Install Template
+
+[Spin templates](https://www.fermyon.com/blog/managing-spin-templates-and-plugins) allow a Spin developer to quickly create the skeleton of an application or component, ready for the application logic to be filled in. As part of this repo, a new template is created to help build applications which make use of MQTT as a communication protocol/trigger.
+
+Install MQTT Template:
+
+```bash
+$ spin templates install --git https://github.com/spinkube/spin-trigger-mqtt --upgrade
+
```
Comment (tpmccallum) [inline content/en/docs/spin-trigger-mqtt/installation/_index.md:35]:
```suggestion
```

File: content/en/docs/spin-trigger-mqtt/installation/_index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+[[component]]
+id = "mqtt-app"
+source = "target/wasm32-wasi/release/mqtt_app.wasm"
+allowed_outbound_hosts = ["mqtt://localhost:1883"]
+[component.trigger]
+topic = "topic"
+qos = "1"
+[component.build]
+command = "cargo build --target wasm32-wasi --release"
+
```
Comment (tpmccallum) [inline content/en/docs/spin-trigger-mqtt/installation/_index.md:74]:
```suggestion
```

File: content/en/docs/spin-trigger-mqtt/installation/_index.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+categories: [guides]
+tags: [plugins, MQTT, spin]
+---
+
+## Installing Spin
+
+If you haven’t already, please go ahead and [install the latest version of Spin](https://developer.fermyon.com/spin/install).
+
+> Upgrading Spin: If you have an older version of Spin, please see the Spin [upgrade](https://developer.fermyon.com/spin/v2/upgrade) page.
+>
```
Comment (tpmccallum) [inline content/en/docs/spin-trigger-mqtt/installation/_index.md:14]:
```suggestion
```

**Review by tpmccallum — COMMENTED** [review]
Removing blank lines

File: content/en/docs/spin-trigger-mqtt/contributing/_index.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  Contributing to the Spin Trigger MQTT
+categories: [contributing]
+tags: [contributing, plugins, mqtt]
+---
+
+## Prerequisites
+
+To compile the plugin from source, you will need to:
+
+* Open the repository in Dev Container or in the pre-configured GitHub [Codespace](https://codespaces.new/spinkube/spin-trigger-mqtt).
```
Comment (endocrimes) [inline content/en/docs/spin-trigger-mqtt/contributing/_index.md:14]:
```suggestion
* Open the repository in a Dev Container or in the pre-configured GitHub [Codespace](https://codespaces.new/spinkube/spin-trigger-mqtt).
```

we should include non-devcontainer instructions for setup too - devcontainers are kinda inherently tied to vscode.

File: content/en/docs/spin-trigger-mqtt/installation/_index.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
+---
+title: Installation
+description: Learn how to install the Spin Trigger MQTT
+weight: 1
+categories: [guides]
+tags: [plugins, MQTT, spin]
```
Comment (endocrimes) [inline content/en/docs/spin-trigger-mqtt/installation/_index.md:6]:
```suggestion
tags: [plugins, MQTT]
```

Spin is somewhat implicit 😅

File: content/en/docs/spin-trigger-mqtt/installation/_index.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
+---
+title: Installation
```
Comment (endocrimes) [inline content/en/docs/spin-trigger-mqtt/installation/_index.md:2]:
```suggestion
title: Local installation
```

The default expectation in spinkube is that you'd be installing on Kubernetes

File: content/en/docs/spin-trigger-mqtt/contributing/_index.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
+---
+title: Contributing
+weight: 5
+description: >
+  Contributing to the Spin Trigger MQTT
+categories: [contributing]
+tags: [contributing, plugins, mqtt]
```
Comment (endocrimes) [inline content/en/docs/spin-trigger-mqtt/contributing/_index.md:7]:
```suggestion
tags: [plugins, mqtt]
```

repeating categories in tags leads to neither really making sense 😅

File: content/en/docs/spin-trigger-mqtt/_index.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+---
+title: spin-trigger-mqtt
+description: Enhance Spin applications by enabling them to integrate between MQTT-based systems
+categories: []
+tags: []
+weight: 80
+---
+
+[Spin Trigger MQTT](https://github.com/spinkube/spin-trigger-mqtt/) is an MQTT trigger designed specifically for Spin. It enables seamless integration between Spin and MQTT-based systems, allowing you to automate workflows and trigger actions based on MQTT messages. With Spin Trigger MQTT, you can easily connect Spin with MQTT brokers and subscribe to topics of interest. When a message is received on a subscribed topic, Spin Trigger MQTT triggers the associated workflow, enabling you to automate processes and streamline your development pipeline. This documentation provides a comprehensive guide on how to set up and use Spin Trigger MQTT effectively, empowering you to leverage the power of MQTT in your Spin workflows.
```
Comment (endocrimes) [inline content/en/docs/spin-trigger-mqtt/_index.md:9]:
```suggestion
[Spin Trigger MQTT](https://github.com/spinkube/spin-trigger-mqtt/) is an MQTT trigger designed specifically for Spin. It enables seamless integration between Spin and MQTT-based systems, allowing you to automate workflows and trigger actions based on MQTT messages. With Spin Trigger MQTT, you can easily connect Spin with MQTT brokers and subscribe to topics of interest. When a message is received on a subscribed topic, Spin Trigger MQTT triggers the associated component. This documentation provides a guide on how to set up and use Spin Trigger MQTT, empowering you to leverage the power of MQTT in your Spin applications.
```

Comment (endocrimes) [reply content/en/docs/spin-trigger-mqtt/_index.md:9]: (we don't have any inherent workflow automation or tooling for business process management - they are much higher level tools than Spin.)

**Review by endocrimes — CHANGES_REQUESTED** [review]

**Comment by tpmccallum** [comment]
Thanks for the inline review comments @endocrimes

File: content/en/docs/spin-trigger-mqtt/contributing/_index.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+## Prerequisites
+
+To compile the plugin from source, you will need to:
+
+* Open the repository in a Dev Container or in the pre-configured GitHub [Codespace](https://codespaces.new/spinkube/spin-trigger-mqtt).
+* Run `make` to build and install the plugin locally.
+* Update `examples/mqtt-app/spin.toml` to reflect your MQTT server details and ensure it's accessible on the network.
+* Run `spin build --up --from examples/mqtt-app/spin.toml` to run the example Spin app.
+* Run `mqttx pub -t 'messages-in01' -h '<mqtt server ip>' -p 1883 -u <user> -P <password> -m 'Hello to  MQTT Spin Component!'` with the hostname and credentials for your server, to publish the message which is then received by Spin app.
```
Comment (endocrimes) [inline content/en/docs/spin-trigger-mqtt/contributing/_index.md:18]:
```suggestion
* Run `mqttx pub -t 'messages-in01' -h '<mqtt server ip>' -p 1883 -u <user> -P <password> -m 'Hello MQTT Spin Component!'` with the hostname and credentials for your server, to publish the message which is then received by Spin app.
```

**Review by endocrimes — APPROVED** [review]
lgtm, one nit

**Comment by tpmccallum** [comment]
Thank you, Danielle!
Got that last inline commit in and am merging now :partying_face:

**Pull request merged**